theano/gpuarray/dnn.py

```diff
@@ -2738,6 +2738,336 @@ def dnn_batch_normalization_test(inputs, gamma, beta, mean, var,
     return result
 
 
+class GpuDnnTransformerDesc(COp):
+    """
+    Descriptor Op for cuDNN Spatial Transformer.
+    """
+    __props__ = ('precision',)
+    params_type = ParamsType(precision=cudnn.cudnnDataType_t)
+
+    def c_headers(self):
+        return ['cudnn.h', 'cudnn_helper.h']
+
+    def c_header_dirs(self):
+        header_dirs = [os.path.dirname(__file__)]
+        if config.dnn.include_path:
+            header_dirs += [config.dnn.include_path]
+        return header_dirs
+
+    def c_libraries(self):
+        return ['cudnn']
+
+    def c_lib_dirs(self):
+        lib_dirs = []
+        if config.dnn.library_path:
+            lib_dirs += [config.dnn.library_path]
+        return lib_dirs
+
+    def do_constant_folding(self, node):
+        return False
+
+    def __init__(self, precision=theano.config.floatX):
+        COp.__init__(self, ["c_code/dnn_sptf_desc.c"], "APPLY_SPECIFIC(dnn_sptf_desc)")
+        assert cudnn.cudnnDataType_t.has_alias(precision)
+        self.precision = precision
```

Comment:
I've removed the descriptor from the implementation, now it is allocated inside the C code of each Op, following the comments on the interface from @notoraptor . In the case of ``GpuDnnTransformerGradT``, it is sufficient to set the number of channels in the descriptor to 1, because it is not a relevant information to that Op.